src/main/flight/mixer.c

```diff
@@ -269,83 +273,121 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
+        signRoll = 0;
+        signPitch = 0;
+    } else {
+        // If pitch/roll dominant, disable yaw
+        signYaw = 0;
+    }
+
+    const float cosPhi = (stickDeflectionLength > 0) ? (stickDeflectionPitchAbs + stickDeflectionRollAbs) / (sqrtf(2.0f) * stickDeflectionLength) : 0;
+    const float cosThreshold = sqrtf(3.0f)/2.0f; // cos(PI/6.0f)
+
+    if (cosPhi < cosThreshold) {
+        // Enforce either roll or pitch exclusively, if not on diagonal
+        if (stickDeflectionRollAbs > stickDeflectionPitchAbs) {
             signPitch = 0;
         } else {
-            // If pitch/roll dominant, disable yaw
-            signYaw = 0;
+            signRoll = 0;
         }
+    }
 
-        const float cosPhi = (stickDeflectionLength > 0) ? (stickDeflectionPitchAbs + stickDeflectionRollAbs) / (sqrtf(2.0f) * stickDeflectionLength) : 0;
-        const float cosThreshold = sqrtf(3.0f)/2.0f; // cos(PI/6.0f)
+    // Calculate flipPower from stick deflection with a reasonable amount of stick deadband
+    float flipPower = stickDeflectionLength > CRASH_FLIP_STICK_MINF ? stickDeflectionLength : 0.0f;
+
+    // calculate flipPower attenuators
+    float flipRateAttenuator = 1.0f;
+    float flipAttitudeAttenuator = 1.0f;
+    const float flipRateLimit = mixerConfig()->crashflip_rate * 10.0f; // eg 35 = no power by 350 deg/s
+
+    // disable both attenuators if the user's crashflip_rate is zero
+    if (flipRateLimit > 0) {
+#ifdef USE_ACC
+        // Calculate the attitude-based attenuator (requires Acc)
+        // with Acc, flipAttitudeAttenuator will be zero after approx 90 degree rotation
+        // hence motors will be stopped while attitude remains more than ~90 degrees from initial attitude
+        // if flipping on the ground, that's exactly what we want; auto-off once flipped.
+        // if stuck in a tree, and is still stuck despite rotating...
+        // re-initialisation of crashFlip mode will be required to be able to drive the motors again
+        // without Acc, the user must manually center the stick, or exit crashflip mode, or disarm, to stop the motors
+        if (sensors(SENSOR_ACC)) {
+            const float tiltAngle = getCosTiltAngle();  // -1 if inverted, 0 when 90°, 1 when flat and upright
+            if (isFirstTiltAngleRead) {
+                tiltAngleAtStart = tiltAngle;
+                isFirstTiltAngleRead = false;
+            }
+            flipAttitudeAttenuator = fmaxf(1.0f - fabsf(tiltAngleAtStart - tiltAngle), 0.0f);
+        }
+#endif // USE_ACC
+        // Calculate attenuation factor for rate of rotation
+        // get the gyro rate for the fastest axis, probably doesn't matter which one
+        // if driving roll or pitch, quad usually turns on that axis, but if one motor sticks, could be a diagonal rotation
+        // if driving diagonally, the turn could be either roll or pitch
+        // if driving yaw, typically one motor sticks, and the quad yaws only a little then flips diagonally
```

Comment:
I appreciate that too many comments are annoying while reviewing.
However, I would like to retain them since:
- they explain why I ended up using the max turn rate on any axis, rather than checking only the turn rate on the controlled axis, which initially seemed to be the way to go
- once the review, and this discussion, is long forgotten, an explanation as to why this approach to this problem was chosen will help people who subsequently want to improve the code
- considerable testing time was required to learn how the quad behaved to the different inputs, and the results are relevant to the code
- by adding comments explaining those findings, users can know what they should expect the behaviour to be for different inputs